sql/analyzer/aliases.go

```diff
@@ -54,7 +54,7 @@ func getTableAliases(n sql.Node, scope *Scope) (TableAliases, error) {
 	var aliasFn func(node sql.Node) bool
 	var analysisErr error
 	var recScope *Scope
-	if scope != nil {
+	if scope != nil && len(scope.Schema()) > 0 {
```

Comment:
Probably makes sense to shorten all these checks to only consider scope.Schema()

You could also introduce an Empty() method for scope